packages/eslint-plugin/tests/rules/no-misused-spread.test.ts

```diff
@@ -0,0 +1,735 @@
+        const o = { ...getSet() };
+      `,
+      errors: [
+        {
+          messageId: 'noSpreadInObject',
+          data: {
+            type: 'Set',
+          },
+          line: 3,
+          column: 21,
+          endColumn: 32,
+        },
+      ],
+    },
+
+    {
+      code: `
+        const o = {
+          ...new Map([
+            ['test-1', 1],
+            ['test-2', 2],
+          ]),
+        };
+      `,
+      errors: [
+        {
+          messageId: 'noSpreadInObject',
+          data: {
+            type: 'Map',
+          },
+          line: 3,
+          column: 11,
+          endLine: 6,
+          endColumn: 13,
+        },
+      ],
+    },
+
+    {
+      code: `
+        const map = new Map([
+          ['test-1', 1],
+          ['test-2', 2],
+        ]);
+
+        const o = { ...map };
+      `,
+      errors: [
+        {
+          messageId: 'noSpreadInObject',
+          data: {
+            type: 'Map',
+          },
+          line: 7,
+          column: 21,
+          endColumn: 27,
+        },
+      ],
+    },
+
+    {
+      code: `
+        declare const map: Map<string, number>;
+        const o = { ...map };
+      `,
+      errors: [
+        {
+          messageId: 'noSpreadInObject',
+          data: {
+            type: 'Map',
+          },
+          line: 3,
+          column: 21,
+          endColumn: 27,
+        },
+      ],
+    },
+
+    {
+      code: `
+        declare function getMap(): Map<string, number>;
+        const o = { ...getMap() };
+      `,
+      errors: [
+        {
+          messageId: 'noSpreadInObject',
+          data: {
+            type: 'Map',
+          },
+          line: 3,
+          column: 21,
+          endColumn: 32,
+        },
+      ],
+    },
+
+    {
+      code: `
+        function f() {}
+
+        const o = { ...f };
+      `,
+      errors: [
+        {
+          messageId: 'noFunctionSpreadInObject',
+          line: 4,
+          column: 21,
+          endColumn: 25,
+        },
+      ],
+    },
+
+    {
+      code: `
+        const f = () => {};
+
+        const o = { ...f };
+      `,
+      errors: [
+        {
+          messageId: 'noFunctionSpreadInObject',
+          line: 4,
+          column: 21,
+          endColumn: 25,
+        },
+      ],
+    },
+
+    {
+      code: `
+        declare function f(): void;
+
+        const o = { ...f };
+      `,
+      errors: [
+        {
+          messageId: 'noFunctionSpreadInObject',
+          line: 4,
+          column: 21,
+          endColumn: 25,
+        },
+      ],
+    },
+
+    {
+      // TODO: Make this test pass.
+      skip: true,
+      code: `
+        declare function getFunction(): () => void;
+
+        const o = { ...getFunction() };
+      `,
+      errors: [
+        {
+          messageId: 'noFunctionSpreadInObject',
+          line: 4,
+          column: 21,
+          endColumn: 37,
+        },
+      ],
+    },
```

Comment:
I can't figure out why the type of `getFunction()` doesn't have the `SymbolFlags.Function`flag set. 
Will be glad for some help here 😄

Reply:
never mind, fixed